File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string,vector<string>>ans;
        vector<vector<string>>g;
        for(int i =  0 ; i < strs.size() ; i++){
            string word = strs[i];
            sort(word.begin(),word.end());
            ans[word].push_back(strs[i]);
        }
        for(auto it:ans){
            g.push_back(it.second);        }
        return g;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        l1=[]
        for i in strs:
            l1.append(''.join(sorted(i)));
        a = []
        for i in range(len(l1)):
            if l1[i] !=0:
                k=[strs[i]]
                for j in range(i+1,len(l1)):
                    if l1[i]==l1[j]:
                        k.append(strs[j])
                        l1[j]=0
                l1[i]=0
                a.append(k)
        a.sort()
        return a
                                 
